endpoints/openrtb2/auction.go

```diff
@@ -2031,6 +2032,181 @@ func setAuctionTypeImplicitly(r *openrtb_ext.RequestWrapper) {
 	}
 }
 
+// (100);v=chrome.1:1:20, (200 400);v=chrome.1:1:40, (300);v=chrome.1:1:60, ();p=P000000000
+func setSecBrowsingTopcisImplicitly(httpReq *http.Request, r *openrtb_ext.RequestWrapper, cfg *config.Configuration) {
+	secBrowsingTopics := httpReq.Header.Get("Sec-Browsing-Topics")
+	if secBrowsingTopics == "" {
+		return
+	}
+
+	// host must configure privacy sandbox
+	if cfg == nil || cfg.Auction.PrivacySandbox.TopicsDomain == "" {
+		return
+	}
+	topicsDomain := cfg.Auction.PrivacySandbox.TopicsDomain
+
+	// segtax-segclass-name-segIds for each lookup against data in request.user.data
+	headerUserData := map[int]map[string]map[string]map[string]struct{}{}
+	// each field is seperated by ,
+	secBrowsingTopicsArr := strings.Split(secBrowsingTopics, ",")
+	selectedFields := 0
+	for _, seg := range secBrowsingTopicsArr {
+		// max 10 fields allowed in header value
+		if selectedFields >= 10 {
+			break
+		}
+
+		seg = strings.TrimSpace(seg)
+		if seg == "" {
+			continue
+		}
+
+		// ();p= is a field to indicated end
+		if strings.HasPrefix(seg, "();p=") {
+			continue
+		}
+
+		// each field has format (segIds);v=browser_version:taxanomy:model_version
+		segment := strings.Split(seg, ";")
+		if len(segment) != 2 {
+			continue
+		}
+
+		// segment[0] is (segIds)
+		// all segIds for a fields should be in () and there should be atleast be one valid segId
+		segmentsIds := strings.TrimSpace(segment[0])
+		if len(segmentsIds) < 3 || segmentsIds[0] != '(' || segmentsIds[len(segmentsIds)-1] != ')' {
+			continue
+		}
+		segmentsIds = strings.TrimLeft(segmentsIds, "(")
+		segmentsIds = strings.TrimRight(segmentsIds, ")")
```

Comment:
You already check the first and last character, so you can use slice instead of the more expensive trim:

```
segmentsIds  = segmentsIds[1:len(segmentsIds)-1]
```

Reply:
done